helix-core/src/test/java/org/apache/helix/integration/manager/TestParticipantManager.java

```diff
@@ -114,6 +118,50 @@ public void simpleIntegrationTest() throws Exception {
     Assert.assertNull(accessor.getProperty(keyBuilder.controllerLeader()));
   }
 
+  @Test
+  public void simpleIntegrationTestNeg() throws Exception {
+
+    TestHelper.setupCluster(clusterName, ZK_ADDR, 12918, // participant port
+        "localhost", // participant name prefix
+        "TestDB", // resource name prefix
+        1, // resources
+        4, // partitions per resource
+        1, // number of nodes
+        1, // replicas
+        "MasterSlave", true); // do rebalance
+
+    ConfigAccessor configAccessor = new ConfigAccessor(_gZkClient);
+    ClusterConfig clusterConfig = configAccessor.getClusterConfig(clusterName);
+    clusterConfig.getRecord()
+        .setListField(ClusterConfig.ClusterConfigProperty.INSTANCE_CAPACITY_KEYS.name(),
+            new ArrayList<>());
+    clusterConfig.setTopologyAwareEnabled(true);
+    clusterConfig.setTopology("/Rack/Sub-Rack/Host/Instance");
+    clusterConfig.setFaultZoneType("Host");
+    configAccessor.setClusterConfig(clusterName, clusterConfig);
+
+
+    String instanceName = "localhost_12918";
+    HelixManager participant =
+        new ZKHelixManager(clusterName, instanceName , InstanceType.PARTICIPANT, ZK_ADDR);
+    participant.getStateMachineEngine().registerStateModelFactory("MasterSlave",
+        new MockMSModelFactory());
+    // We are expecting an IllegalArgumentException since the domain is not set.
+    try {
+      participant.connect();
```

Comment:
Assert fail here to protect the test. Otherwise if expecting exception is not throw, the test will still pass. It should fail.